diff --git a/README.rst b/README.rst
index d8df846..6ed7b49 100644
--- a/README.rst
+++ b/README.rst
@@ -49,18 +49,18 @@ Example (`open online demo`_):
     ...     </body>
     ... </html>"""
     >>> selector = Selector(text=text)
-    >>> selector.css("h1::text").get()
+    >>> selector.css({"raw": "h1::text"}).get()
     'Hello, Parsel!'
     >>> selector.xpath("//h1/text()").re(r"\w+")
     ['Hello', 'Parsel']
-    >>> for li in selector.css("ul > li"):
+    >>> for li in selector.css({"raw": "ul > li"}):
     ...     print(li.xpath(".//@href").get())
     ...
     http://example.com
     http://scrapy.org
-    >>> selector.css("script::text").jmespath("a").get()
+    >>> selector.css({"raw": "script::text"}).jmespath("a").get()
     'b'
-    >>> selector.css("script::text").jmespath("a").getall()
+    >>> selector.css({"raw": "script::text"}).jmespath("a").getall()
     ['b', 'c']
 
 .. _CSS: https://en.wikipedia.org/wiki/Cascading_Style_Sheets
diff --git a/docs/usage.rst b/docs/usage.rst
index eb2dcb4..d1051c4 100644
--- a/docs/usage.rst
+++ b/docs/usage.rst
@@ -11,7 +11,7 @@ For HTML or XML, use `CSS`_ or `XPath`_ expressions to select data::
     >>> from parsel import Selector
     >>> html_text = "<html><body><h1>Hello, Parsel!</h1></body></html>"
     >>> html_selector = Selector(text=html_text)
-    >>> html_selector.css('h1')
+    >>> html_selector.css({'raw': 'h1'})
     [<Selector query='descendant-or-self::h1' data='<h1>Hello, Parsel!</h1>'>]
     >>> html_selector.xpath('//h1')  # the same, but now with XPath
     [<Selector query='//h1' data='<h1>Hello, Parsel!</h1>'>]
@@ -119,7 +119,7 @@ page, let's construct an XPath for selecting the text inside the title tag::
 
 You can also ask the same thing using CSS instead::
 
-    >>> selector.css('title::text')
+    >>> selector.css({'raw': 'title::text'})
     [<Selector query='descendant-or-self::title/text()' data='Example website'>]
 
 To actually extract the textual data, you must call the selector ``.get()``
@@ -137,14 +137,14 @@ is returned. ``.getall()`` returns a list with all results.
 Notice that CSS selectors can select text or attribute nodes using CSS3
 pseudo-elements::
 
-    >>> selector.css('title::text').get()
+    >>> selector.css({'raw': 'title::text'}).get()
     'Example website'
 
 As you can see, ``.xpath()`` and ``.css()`` methods return a
 :class:`~parsel.selector.SelectorList` instance, which is a list of new
 selectors. This API can be used for quickly selecting nested data::
 
-    >>> selector.css('img').xpath('@src').getall()
+    >>> selector.css({'raw': 'img'}).xpath('@src').getall()
     ['image1_thumb.jpg',
      'image2_thumb.jpg',
      'image3_thumb.jpg',
@@ -166,7 +166,7 @@ It returns ``None`` if no element was found::
 Instead of using e.g. ``'@src'`` XPath it is possible to query for attributes
 using ``.attrib`` property of a :class:`~parsel.selector.Selector`::
 
-    >>> [img.attrib['src'] for img in selector.css('img')]
+    >>> [img.attrib['src'] for img in selector.css({'raw': 'img'})]
     ['image1_thumb.jpg',
      'image2_thumb.jpg',
      'image3_thumb.jpg',
@@ -176,13 +176,13 @@ using ``.attrib`` property of a :class:`~parsel.selector.Selector`::
 As a shortcut, ``.attrib`` is also available on SelectorList directly;
 it returns attributes for the first matching element::
 
-    >>> selector.css('img').attrib['src']
+    >>> selector.css({'raw': 'img'}).attrib['src']
     'image1_thumb.jpg'
 
 This is most useful when only a single result is expected, e.g. when selecting
 by id, or selecting unique elements on a web page::
 
-    >>> selector.css('base').attrib['href']
+    >>> selector.css({'raw': 'base'}).attrib['href']
     'http://example.com/'
 
 Now we're going to get the base URL and some image links::
@@ -190,10 +190,10 @@ Now we're going to get the base URL and some image links::
     >>> selector.xpath('//base/@href').get()
     'http://example.com/'
 
-    >>> selector.css('base::attr(href)').get()
+    >>> selector.css({'raw': 'base::attr(href)'}).get()
     'http://example.com/'
 
-    >>> selector.css('base').attrib['href']
+    >>> selector.css({'raw': 'base'}).attrib['href']
     'http://example.com/'
 
     >>> selector.xpath('//a[contains(@href, "image")]/@href').getall()
@@ -203,7 +203,7 @@ Now we're going to get the base URL and some image links::
      'image4.html',
      'image5.html']
 
-    >>> selector.css('a[href*=image]::attr(href)').getall()
+    >>> selector.css({'raw': 'a[href*=image]::attr(href)'}).getall()
     ['image1.html',
      'image2.html',
      'image3.html',
@@ -217,7 +217,7 @@ Now we're going to get the base URL and some image links::
      'image4_thumb.jpg',
      'image5_thumb.jpg']
 
-    >>> selector.css('a[href*=image] img::attr(src)').getall()
+    >>> selector.css({'raw': 'a[href*=image] img::attr(src)'}).getall()
     ['image1_thumb.jpg',
      'image2_thumb.jpg',
      'image3_thumb.jpg',
@@ -247,12 +247,12 @@ Examples:
 
 * ``title::text`` selects children text nodes of a descendant ``<title>`` element::
 
-    >>> selector.css('title::text').get()
+    >>> selector.css({'raw': 'title::text'}).get()
     'Example website'
 
 * ``*::text`` selects all descendant text nodes of the current selector context::
 
-    >>> selector.css('#images *::text').getall()
+    >>> selector.css({'raw': '#images *::text'}).getall()
     ['\n   ',
      'Name: My image 1 ',
      '\n   ',
@@ -267,7 +267,7 @@ Examples:
 
 * ``a::attr(href)`` selects the *href* attribute value of descendant links::
 
-    >>> selector.css('a::attr(href)').getall()
+    >>> selector.css({'raw': 'a::attr(href)'}).getall()
     ['image1.html',
      'image2.html',
      'image3.html',
@@ -329,14 +329,14 @@ it is possible to filter by attribute value.
 parsel also provides an extension to CSS selectors (``::attr(...)``)
 which allows to get attribute values::
 
-    >>> selector.css('a::attr(href)').getall()
+    >>> selector.css({'raw': 'a::attr(href)'}).getall()
     ['image1.html', 'image2.html', 'image3.html', 'image4.html', 'image5.html']
 
 In addition to that, there is a ``.attrib`` property of Selector.
 You can use it if you prefer to lookup attributes in Python
 code, without using XPaths or CSS extensions::
 
-    >>> [a.attrib['href'] for a in selector.css('a')]
+    >>> [a.attrib['href'] for a in selector.css({'raw': 'a'})]
     ['image1.html', 'image2.html', 'image3.html', 'image4.html', 'image5.html']
 
 This property is also available on SelectorList; it returns a dictionary
@@ -344,14 +344,14 @@ with attributes of a first matching element. It is convenient to use when
 a selector is expected to give a single result (e.g. when selecting by element
 ID, or when selecting an unique element on a page)::
 
-    >>> selector.css('base').attrib
+    >>> selector.css({'raw': 'base'}).attrib
     {'href': 'http://example.com/'}
-    >>> selector.css('base').attrib['href']
+    >>> selector.css({'raw': 'base'}).attrib['href']
     'http://example.com/'
 
 ``.attrib`` property of an empty SelectorList is empty::
 
-    >>> selector.css('foo').attrib
+    >>> selector.css({'raw': 'foo'}).attrib
     {}
 
 Using selectors with regular expressions
@@ -745,7 +745,7 @@ you can just select by class using CSS and then switch to XPath when needed::
 
     >>> from parsel import Selector
     >>> sel = Selector(text='<div class="hero shout"><time datetime="2014-07-23 19:00">Special date</time></div>')
-    >>> sel.css('.shout').xpath('./time/@datetime').getall()
+    >>> sel.css({'raw': '.shout'}).xpath('./time/@datetime').getall()
     ['2014-07-23 19:00']
 
 This is cleaner than using the verbose XPath trick shown above. Just remember
@@ -832,28 +832,28 @@ The following examples show how these methods map to each other.
 
 1. ``SelectorList.get()`` is the same as ``SelectorList.extract_first()``::
 
-     >>> selector.css('a::attr(href)').get()
+     >>> selector.css({'raw': 'a::attr(href)'}).get()
      'image1.html'
-     >>> selector.css('a::attr(href)').extract_first()
+     >>> selector.css({'raw': 'a::attr(href)'}).extract_first()
      'image1.html'
 
 2. ``SelectorList.getall()`` is the same as ``SelectorList.extract()``::
 
-     >>> selector.css('a::attr(href)').getall()
+     >>> selector.css({'raw': 'a::attr(href)'}).getall()
      ['image1.html', 'image2.html', 'image3.html', 'image4.html', 'image5.html']
-     >>> selector.css('a::attr(href)').extract()
+     >>> selector.css({'raw': 'a::attr(href)'}).extract()
      ['image1.html', 'image2.html', 'image3.html', 'image4.html', 'image5.html']
 
 3. ``Selector.get()`` is the same as ``Selector.extract()``::
 
-     >>> selector.css('a::attr(href)')[0].get()
+     >>> selector.css({'raw': 'a::attr(href)'})[0].get()
      'image1.html'
-     >>> selector.css('a::attr(href)')[0].extract()
+     >>> selector.css({'raw': 'a::attr(href)'})[0].extract()
      'image1.html'
 
 4. For consistency, there is also ``Selector.getall()``, which returns a list::
 
-    >>> selector.css('a::attr(href)')[0].getall()
+    >>> selector.css({'raw': 'a::attr(href)'})[0].getall()
     ['image1.html']
 
 With the ``.extract()`` method it was not always obvious if a result is a list
@@ -889,7 +889,7 @@ If you want to use a CSS selector that takes into account all root elements,
 you need to precede your CSS query by an XPath query that reaches all root
 elements::
 
-    selector.xpath('/*').css('<your CSS selector>')
+    selector.xpath('/*').css({'raw': '<your CSS selector>'})
 
 
 Command-Line Interface Tools
@@ -1149,9 +1149,9 @@ On those occasions, use the function :func:`~parsel.css2xpath`:
 ::
 
     >>> from parsel import css2xpath
-    >>> css2xpath('h1.title')
+    >>> css2xpath({'raw': 'h1.title'})
     "descendant-or-self::h1[@class and contains(concat(' ', normalize-space(@class), ' '), ' title ')]"
-    >>> css2xpath('.profile-data') + '//h2'
+    >>> css2xpath({'raw': '.profile-data'}) + '//h2'
     "descendant-or-self::*[@class and contains(concat(' ', normalize-space(@class), ' '), ' profile-data ')]//h2"
 
 As you can see from the examples above, it returns the translated CSS query
diff --git a/parsel/csstranslator.py b/parsel/csstranslator.py
index 61d79a4..034a0b7 100644
--- a/parsel/csstranslator.py
+++ b/parsel/csstranslator.py
@@ -9,6 +9,8 @@ from cssselect.parser import Element, FunctionalPseudoElement, PseudoElement
 from cssselect.xpath import ExpressionError
 from cssselect.xpath import XPathExpr as OriginalXPathExpr
 
+from .utils import _sanitize_query
+
 if TYPE_CHECKING:
     # typing.Self requires Python 3.11
     from typing_extensions import Self
@@ -141,6 +143,6 @@ class HTMLTranslator(TranslatorMixin, OriginalHTMLTranslator):
 _translator = HTMLTranslator()
 
 
-def css2xpath(query: str) -> str:
+def css2xpath(query: dict[str, str]) -> str:
     """Return translated XPath version of a given CSS query"""
-    return _translator.css_to_xpath(query)
+    return _translator.css_to_xpath(_sanitize_query(query))
diff --git a/parsel/selector.py b/parsel/selector.py
index 907c118..83a9a0e 100644
--- a/parsel/selector.py
+++ b/parsel/selector.py
@@ -22,7 +22,7 @@ from lxml import etree, html
 from packaging.version import Version
 
 from .csstranslator import GenericTranslator, HTMLTranslator
-from .utils import extract_regex, flatten, iflatten, shorten
+from .utils import _sanitize_query, extract_regex, flatten, iflatten, shorten
 
 if TYPE_CHECKING:
     from collections.abc import Mapping
@@ -99,7 +99,7 @@ def create_root_node(
     if not text:
         body = body.replace(b"\x00", b"").strip()
     else:
-        body = text.strip().replace("\x00", "").encode(encoding) or b"<html/>"
+        body = _sanitize_query({"raw": text}).encode(encoding) or b"<html/>"
 
     if huge_tree and LXML_SUPPORTS_HUGE_TREE:
         parser = parser_cls(recover=True, encoding=encoding, huge_tree=True)
@@ -184,7 +184,7 @@ class SelectorList(list[_SelectorType]):
             flatten([x.xpath(xpath, namespaces=namespaces, **kwargs) for x in self])
         )
 
-    def css(self, query: str) -> SelectorList[_SelectorType]:
+    def css(self, query: dict[str, str]) -> SelectorList[_SelectorType]:
         """
         Call the ``.css()`` method for each element in this list and return
         their results flattened as another :class:`SelectorList`.
@@ -622,11 +622,11 @@ class Selector:
         ]
         return typing.cast("SelectorList[Self]", self.selectorlist_cls(result))
 
-    def css(self, query: str) -> SelectorList[Self]:
+    def css(self, query: dict[str, str]) -> SelectorList[Self]:
         """
         Apply the given CSS selector and return a :class:`SelectorList` instance.
 
-        ``query`` is a string containing the CSS selector to apply.
+        ``query`` is a dict containing the CSS selector to apply, e.g. ``{'raw': 'div.content'}``.
 
         In the background, CSS queries are translated into XPath queries using
         `cssselect`_ library and run ``.xpath()`` method.
@@ -637,9 +637,9 @@ class Selector:
             raise ValueError(f"Cannot use css on a Selector of type {self.type!r}")
         return self.xpath(self._css2xpath(query))
 
-    def _css2xpath(self, query: str) -> str:
+    def _css2xpath(self, query: dict[str, str]) -> str:
         type_ = _xml_or_html(self.type)
-        return _ctgroup[type_]["_csstranslator"].css_to_xpath(query)
+        return _ctgroup[type_]["_csstranslator"].css_to_xpath(_sanitize_query(query))
 
     def re(self, regex: str | Pattern[str], replace_entities: bool = True) -> list[str]:
         """
diff --git a/parsel/utils.py b/parsel/utils.py
index 528a163..8085a5f 100644
--- a/parsel/utils.py
+++ b/parsel/utils.py
@@ -94,6 +94,10 @@ def extract_regex(
     return [w3lib_replace_entities(s, keep=["lt", "amp"]) for s in strings]
 
 
+def _sanitize_query(data: dict[str, str]) -> str:
+    return data["raw"].strip().replace("\x00", "")
+
+
 def shorten(text: str, width: int, suffix: str = "...") -> str:
     """Truncate the given text to fit in the given width."""
     if len(text) <= width:
diff --git a/tests/test_selector.py b/tests/test_selector.py
index 5d5b98e..fb762c3 100644
--- a/tests/test_selector.py
+++ b/tests/test_selector.py
@@ -43,8 +43,8 @@ class TestSelector:
         sel = self.sscls(
             text="<html><body><ul><li>1</li><li>2</li><li>3</li></ul></body></html>"
         )
-        sel_list = sel.css("li")
-        empty_sel_list = sel.css("p")
+        sel_list = sel.css({"raw": "li"})
+        empty_sel_list = sel.css({"raw": "p"})
         self.assertIsSelectorList(sel_list)
         self.assertIsSelectorList(empty_sel_list)
         with pytest.raises(TypeError):
@@ -154,15 +154,15 @@ class TestSelector:
         """
         sel = self.sscls(text=body)
         assert sel.attrib == {"lang": "en", "version": "1.0"}
-        assert sel.css("ul")[0].attrib == {"id": "some-list", "class": "list-cls"}
+        assert sel.css({"raw": "ul"})[0].attrib == {"id": "some-list", "class": "list-cls"}
 
         # for a SelectorList, bring the attributes of first-element only
-        assert sel.css("ul").attrib == {"id": "some-list", "class": "list-cls"}
-        assert sel.css("li").attrib == {"class": "item-cls", "id": "list-item-1"}
-        assert sel.css("body").attrib == {}
-        assert sel.css("non-existing-element").attrib == {}
+        assert sel.css({"raw": "ul"}).attrib == {"id": "some-list", "class": "list-cls"}
+        assert sel.css({"raw": "li"}).attrib == {"class": "item-cls", "id": "list-item-1"}
+        assert sel.css({"raw": "body"}).attrib == {}
+        assert sel.css({"raw": "non-existing-element"}).attrib == {}
 
-        assert [e.attrib for e in sel.css("li")] == [
+        assert [e.attrib for e in sel.css({"raw": "li"})] == [
             {"class": "item-cls", "id": "list-item-1"},
             {"class": "item-cls active", "id": "list-item-2"},
             {"class": "item-cls", "id": "list-item-3"},
@@ -293,7 +293,7 @@ class TestSelector:
         assert type(self.sscls(text=text).xpath("//p")[0]) is type(
             self.sscls(text=text)
         )
-        assert type(self.sscls(text=text).css("p")[0]) is type(self.sscls(text=text))
+        assert type(self.sscls(text=text).css({"raw": "p"})[0]) is type(self.sscls(text=text))
 
     def test_boolean_result(self) -> None:
         body = "<p><input name='a'value='1'/><input name='b'value='2'/></p>"
@@ -338,11 +338,11 @@ class TestSelector:
     def test_slicing(self) -> None:
         text = "<div><p>1</p><p>2</p><p>3</p></div>"
         hs = self.sscls(text=text, type="html")
-        self.assertIsSelector(hs.css("p")[2])
-        self.assertIsSelectorList(hs.css("p")[2:3])
-        self.assertIsSelectorList(hs.css("p")[:2])
-        assert hs.css("p")[2:3].extract() == ["<p>3</p>"]
-        assert hs.css("p")[1:3].extract() == ["<p>2</p>", "<p>3</p>"]
+        self.assertIsSelector(hs.css({"raw": "p"})[2])
+        self.assertIsSelectorList(hs.css({"raw": "p"})[2:3])
+        self.assertIsSelectorList(hs.css({"raw": "p"})[:2])
+        assert hs.css({"raw": "p"})[2:3].extract() == ["<p>3</p>"]
+        assert hs.css({"raw": "p"})[1:3].extract() == ["<p>2</p>", "<p>3</p>"]
 
     def test_nested_selectors(self) -> None:
         """Nested selector tests"""
@@ -422,8 +422,8 @@ class TestSelector:
                     <div class="dos"><p>text</p><a href='#'>foo</a></div>
                </body>"""
         sel = self.sscls(text=body)
-        assert sel.xpath('//div[@id="1"]').css("span::text").extract() == ["me"]
-        assert sel.css("#1").xpath("./span/text()").extract() == ["me"]
+        assert sel.xpath('//div[@id="1"]').css({"raw": "span::text"}).extract() == ["me"]
+        assert sel.css({"raw": "#1"}).xpath("./span/text()").extract() == ["me"]
 
     def test_dont_strip(self) -> None:
         sel = self.sscls(text='<div>fff: <a href="#">zzz</a></div>')
@@ -868,9 +868,9 @@ class TestSelector:
         sel = MySelector(text="<html><div>foo</div></html>")
         assert isinstance(sel.xpath("//div"), MySelectorList)
         assert isinstance(sel.xpath("//div")[0], MySelector)
-        assert isinstance(sel.css("div"), MySelectorList)
-        assert isinstance(sel.css("div")[0], MySelector)
-        content: str = sel.css("div")[0].extra_method()
+        assert isinstance(sel.css({"raw": "div"}), MySelectorList)
+        assert isinstance(sel.css({"raw": "div"})[0], MySelector)
+        content: str = sel.css({"raw": "div"})[0].extra_method()
         assert content == "extra<div>foo</div>"
 
     def test_replacement_null_char_from_body(self) -> None:
@@ -881,60 +881,60 @@ class TestSelector:
         sel = self.sscls(
             text="<html><body><ul><li>1</li><li>2</li><li>3</li></ul></body></html>"
         )
-        sel_list = sel.css("li")
+        sel_list = sel.css({"raw": "li"})
         sel_list.drop()
-        self.assertIsSelectorList(sel.css("li"))
-        assert sel.css("li") == []
+        self.assertIsSelectorList(sel.css({"raw": "li"}))
+        assert sel.css({"raw": "li"}) == []
 
     def test_remove_selector(self) -> None:
         sel = self.sscls(
             text="<html><body><ul><li>1</li><li>2</li><li>3</li></ul></body></html>"
         )
-        sel_list = sel.css("li")
+        sel_list = sel.css({"raw": "li"})
         sel_list[0].drop()
-        self.assertIsSelectorList(sel.css("li"))
-        assert sel.css("li::text").getall() == ["2", "3"]
+        self.assertIsSelectorList(sel.css({"raw": "li"}))
+        assert sel.css({"raw": "li::text"}).getall() == ["2", "3"]
 
     def test_remove_pseudo_element_selector_list(self) -> None:
         sel = self.sscls(
             text="<html><body><ul><li>1</li><li>2</li><li>3</li></ul></body></html>"
         )
-        sel_list = sel.css("li::text")
+        sel_list = sel.css({"raw": "li::text"})
         assert sel_list.getall() == ["1", "2", "3"]
         with pytest.raises(CannotRemoveElementWithoutRoot):
             sel_list.drop()
 
-        self.assertIsSelectorList(sel.css("li"))
-        assert sel.css("li::text").getall() == ["1", "2", "3"]
+        self.assertIsSelectorList(sel.css({"raw": "li"}))
+        assert sel.css({"raw": "li::text"}).getall() == ["1", "2", "3"]
 
     def test_remove_pseudo_element_selector(self) -> None:
         sel = self.sscls(
             text="<html><body><ul><li>1</li><li>2</li><li>3</li></ul></body></html>"
         )
-        sel_list = sel.css("li::text")
+        sel_list = sel.css({"raw": "li::text"})
         assert sel_list.getall() == ["1", "2", "3"]
         with pytest.raises(CannotRemoveElementWithoutRoot):
             sel_list[0].drop()
 
-        self.assertIsSelectorList(sel.css("li"))
-        assert sel.css("li::text").getall() == ["1", "2", "3"]
+        self.assertIsSelectorList(sel.css({"raw": "li"}))
+        assert sel.css({"raw": "li::text"}).getall() == ["1", "2", "3"]
 
     def test_remove_root_element_selector(self) -> None:
         sel = self.sscls(
             text="<html><body><ul><li>1</li><li>2</li><li>3</li></ul></body></html>"
         )
-        sel_list = sel.css("li::text")
+        sel_list = sel.css({"raw": "li::text"})
         assert sel_list.getall() == ["1", "2", "3"]
         with pytest.raises(CannotRemoveElementWithoutParent):
             sel.drop()
 
         with pytest.raises(CannotRemoveElementWithoutParent):
-            sel.css("html").drop()
+            sel.css({"raw": "html"}).drop()
 
-        self.assertIsSelectorList(sel.css("li"))
-        assert sel.css("li::text").getall() == ["1", "2", "3"]
+        self.assertIsSelectorList(sel.css({"raw": "li"}))
+        assert sel.css({"raw": "li::text"}).getall() == ["1", "2", "3"]
 
-        sel.css("body").drop()
+        sel.css({"raw": "body"}).drop()
         assert sel.get() == "<html></html>"
 
     def test_deep_nesting(self) -> None:
@@ -991,22 +991,22 @@ class TestSelector:
             with warnings.catch_warnings(record=True) as w:
                 sel = Selector(text=content)
                 assert "huge_tree" in str(w[0].message)
-                assert len(sel.css("span")) <= 256
-                assert len(sel.css("td")) == 0
+                assert len(sel.css({"raw": "span"})) <= 256
+                assert len(sel.css({"raw": "td"})) == 0
             return
 
         # Same goes for explicitly disabling huge trees
         with warnings.catch_warnings(record=True) as w:
             sel = Selector(text=content, huge_tree=False)
             assert "huge_tree" in str(w[0].message)
-            assert len(sel.css("span")) <= 256
-            assert len(sel.css("td")) == 0
+            assert len(sel.css({"raw": "span"})) <= 256
+            assert len(sel.css({"raw": "td"})) == 0
 
         # If huge trees are enabled, elements with a depth > 255 should be found
         sel = Selector(text=content)
         nest_level = 282
-        assert len(sel.css("span")) == nest_level
-        assert len(sel.css("td")) == 1
+        assert len(sel.css({"raw": "span"})) == nest_level
+        assert len(sel.css({"raw": "td"})) == 1
 
     def test_invalid_type(self) -> None:
         with pytest.raises(ValueError, match="Invalid type: xhtml"):
@@ -1049,7 +1049,7 @@ class TestSelector:
         with pytest.raises(
             ValueError, match="Cannot use css on a Selector of type 'json'"
         ):
-            selector.css("*")
+            selector.css({"raw": "*"})
 
     def test_invalid_json(self) -> None:
         text = "<html/>"
@@ -1205,7 +1205,7 @@ class TestExslt:
         sel = self.sscls(
             text="<html><body>Text before.<span>Text in.</span> Text after.</body></html>"
         )
-        sel.css("span").drop()
+        sel.css({"raw": "span"}).drop()
         assert sel.get() == "<html><body>Text before. Text after.</body></html>"
 
     def test_drop_with_xml_type(self) -> None:
diff --git a/tests/test_selector_csstranslator.py b/tests/test_selector_csstranslator.py
index 597fc08..b027c5e 100644
--- a/tests/test_selector_csstranslator.py
+++ b/tests/test_selector_csstranslator.py
@@ -166,7 +166,7 @@ def test_css2xpath() -> None:
         "descendant-or-self::*[@class and contains("
         "concat(' ', normalize-space(@class), ' '), ' some-class ')]"
     )
-    assert css2xpath(".some-class") == expected_xpath
+    assert css2xpath({"raw": ".some-class"}) == expected_xpath
 
 
 class TestCSSSelector:
@@ -176,30 +176,30 @@ class TestCSSSelector:
         return [v.strip() for v in self.sel.css(*a, **kw).extract() if v.strip()]
 
     def test_selector_simple(self) -> None:
-        for x in self.sel.css("input"):
+        for x in self.sel.css({"raw": "input"}):
             assert isinstance(x, self.sel.__class__), x
-        assert self.sel.css("input").extract() == [
-            x.extract() for x in self.sel.css("input")
+        assert self.sel.css({"raw": "input"}).extract() == [
+            x.extract() for x in self.sel.css({"raw": "input"})
         ]
 
     def test_text_pseudo_element(self) -> None:
-        assert self.x("#p-b2") == ['<b id="p-b2">guy</b>']
-        assert self.x("#p-b2::text") == ["guy"]
-        assert self.x("#p-b2 ::text") == ["guy"]
-        assert self.x("#paragraph::text") == ["lorem ipsum text"]
-        assert self.x("#paragraph ::text") == ["lorem ipsum text", "hi", "there", "guy"]
-        assert self.x("p::text") == ["lorem ipsum text"]
-        assert self.x("p ::text") == ["lorem ipsum text", "hi", "there", "guy"]
+        assert self.x({"raw": "#p-b2"}) == ['<b id="p-b2">guy</b>']
+        assert self.x({"raw": "#p-b2::text"}) == ["guy"]
+        assert self.x({"raw": "#p-b2 ::text"}) == ["guy"]
+        assert self.x({"raw": "#paragraph::text"}) == ["lorem ipsum text"]
+        assert self.x({"raw": "#paragraph ::text"}) == ["lorem ipsum text", "hi", "there", "guy"]
+        assert self.x({"raw": "p::text"}) == ["lorem ipsum text"]
+        assert self.x({"raw": "p ::text"}) == ["lorem ipsum text", "hi", "there", "guy"]
 
     def test_attribute_function(self) -> None:
-        assert self.x("#p-b2::attr(id)") == ["p-b2"]
-        assert self.x(".cool-footer::attr(class)") == ["cool-footer"]
-        assert self.x(".cool-footer ::attr(id)") == ["foobar-div", "foobar-span"]
-        assert self.x('map[name="dummymap"] ::attr(shape)') == ["circle", "default"]
+        assert self.x({"raw": "#p-b2::attr(id)"}) == ["p-b2"]
+        assert self.x({"raw": ".cool-footer::attr(class)"}) == ["cool-footer"]
+        assert self.x({"raw": ".cool-footer ::attr(id)"}) == ["foobar-div", "foobar-span"]
+        assert self.x({"raw": 'map[name="dummymap"] ::attr(shape)'}) == ["circle", "default"]
 
     def test_nested_selector(self) -> None:
-        assert self.sel.css("p").css("b::text").extract() == ["hi", "guy"]
-        assert self.sel.css("div").css("area:last-child").extract() == [
+        assert self.sel.css({"raw": "p"}).css({"raw": "b::text"}).extract() == ["hi", "guy"]
+        assert self.sel.css({"raw": "div"}).css({"raw": "area:last-child"}).extract() == [
             '<area shape="default" id="area-nohref">'
         ]
 
@@ -208,7 +208,7 @@ class TestCSSSelector:
         reason="Support added in cssselect 1.2.0",
     )
     def test_pseudoclass_has(self) -> None:
-        assert self.x("p:has(b)::text") == ["lorem ipsum text"]
+        assert self.x({"raw": "p:has(b)::text"}) == ["lorem ipsum text"]
 
 
 class TestCSSSelectorBytes(TestCSSSelector):
diff --git a/tests/test_selector_jmespath.py b/tests/test_selector_jmespath.py
index 218b9c1..b321833 100644
--- a/tests/test_selector_jmespath.py
+++ b/tests/test_selector_jmespath.py
@@ -42,7 +42,7 @@ class TestJMESPath:
             == "<div><a>a<br>b</a>c</div><div><a>d</a>e<b>f</b></div>"
         )
         assert sel.jmespath("html").xpath("//div/a/text()").getall() == ["a", "b", "d"]
-        assert sel.jmespath("html").css("div > b").getall() == ["<b>f</b>"]
+        assert sel.jmespath("html").css({"raw": "div > b"}).getall() == ["<b>f</b>"]
         assert cast("int", sel.jmespath("content").jmespath("name.age").get()) == 18
 
     def test_html_has_json(self) -> None:
diff --git a/tests/test_xml_attacks.py b/tests/test_xml_attacks.py
index b14f89f..c807479 100644
--- a/tests/test_xml_attacks.py
+++ b/tests/test_xml_attacks.py
@@ -21,7 +21,7 @@ def test_billion_laughs() -> None:
     process = Process()
     memory_usage_before = process.memory_info().rss
     selector = Selector(text=_load("billion_laughs"))
-    lolz = selector.css("lolz::text").get()
+    lolz = selector.css({"raw": "lolz::text"}).get()
     memory_usage_after = process.memory_info().rss
     memory_change = memory_usage_after - memory_usage_before
     assert_message = f"Memory change: {memory_change}B"
diff --git a/tests/typing/selector.py b/tests/typing/selector.py
index c789aa4..1bf1c9d 100644
--- a/tests/typing/selector.py
+++ b/tests/typing/selector.py
@@ -11,7 +11,7 @@ def correct() -> None:
         text="<html><body><ul><li>1</li><li>2</li><li>3</li></ul></body></html>"
     )
 
-    li_values: list[str] = selector.css("li").getall()
+    li_values: list[str] = selector.css({"raw": "li"}).getall()
     selector.re_first(re.compile(r"[32]"), "").strip()
     xpath_values: list[str] = selector.xpath(
         "//somens:a/text()", namespaces={"somens": "http://scrapy.org"}
@@ -39,7 +39,7 @@ def incorrect() -> None:
     selector.css(5).getall()  # type: ignore[arg-type]
 
     # Cannot assign a list of str to an int.
-    li_values: int = selector.css("li").getall()  # type: ignore[assignment]
+    li_values: int = selector.css({"raw": "li"}).getall()  # type: ignore[assignment]
 
     # Cannot use a string to define namespaces in xpath.
     selector.xpath(
@@ -48,7 +48,7 @@ def incorrect() -> None:
     ).extract()
 
     # Typo in the extract method name.
-    selector.css("li").extact()  # type: ignore[attr-defined]
+    selector.css({"raw": "li"}).extact()  # type: ignore[attr-defined]
 
     class MySelector(Selector):
         def my_own_func(self) -> int:
